aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -256,6 +270,43 @@ module aptos_framework::account {
         account_resource.authentication_key = new_auth_key;
     }
 
+    public entry fun offer_rotation_capability(account: &signer, rotation_capability_signature: vector<u8>, public_key: vector<u8>, intended_receiver_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(intended_receiver_address), error::not_found(EACCOUNT));
+        assert!(
+            vector::length(&public_key) == 32,
+            error::invalid_argument(EMALFORMED_PUBLIC_KEY)
+        );
+        assert!(
+            vector::length(&rotation_capability_signature) == 64,
+            error::invalid_argument(EMALFORMED_PROOF_OF_KNOWLEDGE)
+        );
+
+        let account_resource = borrow_global_mut<Account>(addr);
+
+        let rotation_capability_offer_proof = RotationCapabilityOfferProof {
+            sequence_number: account_resource.sequence_number,
+            intended_receiver_address,
+        };
+
+        assert!(signature::ed25519_verify_t(rotation_capability_signature, public_key, rotation_capability_offer_proof), EINVALID_PROOF_OF_KNOWLEDGE);
+        option::fill(&mut account_resource.rotation_capability_offer.for, intended_receiver_address);
+    }
+
+    public fun accept_rotation_capability(account: &signer, offerer_address: address) acquires Account {
+        let addr = signer::address_of(account);
+        assert!(exists_at(addr) && exists_at(offerer_address), error::not_found(EACCOUNT));
+
+        let account_resource = borrow_global_mut<Account>(offerer_address);
+        assert!(option::contains(&account_resource.rotation_capability_offer.for, &addr), EINVALID_ACCEPT_ROTATION_CAPABILITY);
+
+        let rotation_capability = RotationCapability {
+            account: offerer_address,
+        };
+        move_to(account, rotation_capability);
```

Comment:
just return it, no need to store it, that is something we can do in the test code